File: 202311/1846.go
import "slices"

func maximumElementAfterDecrementingAndRearranging(arr []int) int {
	slices.Sort(arr)
	if arr[0] != 1 {
		arr[0] = 1
	}
	n := len(arr)
	for i := 1; i < n; i++ {
		arr[i] = min(arr[i-1]+1, arr[i])
	}
	return arr[n-1]
}

File: 202606/1846.cpp
class Solution {
public:
  int maximumElementAfterDecrementingAndRearranging(vector<int> &arr) {
    sort(begin(arr), end(arr));
    if (arr[0] != 1)
      arr[0] = 1;
    for (int i = 1; i < arr.size(); ++i) {
      if (arr[i] - arr[i - 1] > 1) {
        arr[i] = arr[i - 1] + 1;
      }
    }
    return arr[arr.size() - 1];
  }
};
